software/follow_ball_depth.py

```diff
@@ -0,0 +1,381 @@
+
+                except:
+                    print('except')
+                    motion_sim3.move(0, 0, 0,100)
+                    state = 'find_ball'
+
+        else : 
+            state = 'find_ball'
+
+
+class Scanning_ball(threading.Thread):
+
+    def __init__(self,name,processor):
+        threading.Thread.__init__(self)
+        self.name = name
+        self.stop = False
+        self.processor = processor
+        self.count = 0
+        self.basket_a = None
+
+    def run(self):
+        print("Starting " + self.name)
+
+    def scan_ball(self):
+        global state 
+        global oponent_basket
+        global target_basket
+        processedData = self.processor.process_frame(aligned_depth=False)
+        
+        if target_basket == 'blue':
+            direction = 1
+        else:
+            direction = -1
+        
+        if state == 'find_ball':
+            motion_sim3.move(0, 0, direction * 3,100)
+            # time.sleep(0.2)
+            # motion_sim3.move(0, 0, 0,100)
+            # time.sleep(0.2)
+            self.count+=1
+
+                # print(self.count)
+                # if oponent_basket == 'blue':
+                #     basket_ = processedData.basket_b
+                # else:
+                #     basket_ = processedData.basket_m
+
+                # try:    
+                #     print(basket_)
+                #     if bakset_.exists :
+                #         print('go to basket')
+                #         motion_sim3.move(3, 0, basket_.x/100,100)
+                #         time.sleep(0.6)
+                #         self.count = 0
+                #     else: 
+                #         motion_sim3.move_turn()
+                #         time.sleep(0.2)
+                #         motion_sim3.move(0, 0, 0,100)
+                #         time.sleep(0.6)
+                # except: 
+                #     motion_sim3.move_turn()
+                #     time.sleep(0.2)
+                #     motion_sim3.move(0, 0, 0,100)
+                #     time.sleep(0.6)
+
+class referee_Thread(threading.Thread):
+
+    def __init__(self,name):
+        threading.Thread.__init__(self)
+        self.rf_command = referee_command.referee_command()
+        self.name = name
+        self.interupt = True
+        self.stop = False
+
+    def run(self):
+        global target_basket
+        global oponent_basket
+        global state 
+        print("Starting " + self.name)
+        try:
+            self.rf_command.connect()
+        except:
+            print("can not connect to server")
+            self.interupt = True
+        
+        while True:
+            msg = json.loads(self.rf_command.listen())
+            if msg.get("signal") == "stop":
+                self.interupt = True
+                state = 'find_ball'
+                print(msg.get("signal"))
+
+            else:
+                for robot in range(len(msg.get("targets"))):
```

Comment:
Try to use a if `msg.get("targets").index(robot_id) `here to find the index of your robots ID.